### ISSUE TITLE
Waterfox Classic: The connection to the server was reset while the page was loading.

### ISSUE THREAD

**Issue by grahamperrin** [MrAlex94/Waterfox#1259]
Bugging some pages on some distributions of Linux. 

From the https://www.reddit.com/comments/dorwpc/-/f6o3a0q/?context=1 part of [Cannot log into my Bank Account with new Waterfox Classic on Linux since update !?](https://www.reddit.com/r/waterfox/comments/dorwpc/cannot_log_into_my_bank_account_with_new_waterfox/):

> **Secure Connection Failed**
> 
> The connection to the server was reset while the page was loading.
> 
> The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
> 
> Please contact the website owners to inform them of this problem.

Resets of this type seem to be consistently reproducible at (for example): 

- https://onlinebanking.mtb.com/Enrollment/Enroll

– with these releases, maybe others, on Lubuntu: 

- 56.2.7
- 56.2.14
- 2019.10 (20191023220600) i.e. `2019.10-classic-1`.

My Lubuntu in a VirtualBox guest is a few months old but regularly updated and upgraded so I guess, other users/testers will be able to reproduce the issue with <strike>19.04 </strike>19.10 from ~~`https://lubuntu.net/downloads/`~~ – for myself, I might boot the image in live mode and aim to make this issue reproducible in the cleanest possible environment. 

Screen recording to follow. It's quite long (sorry) so I'll include it in a separate comment that might be hidden/outdated as things progress. tl;dw from the video: 

- apparent successions of **TLS handshakes**, or attempts to handshake

– part of the unwanted length of the video was me not knowing how to dig deeper (with consoles etc.) in situations such as this. 

Meta, tracking: #538 

### Postscript

Re: https://www.reddit.com/r/Lubuntu/comments/djaiub/-/f6rjvdc/?context=2 please ignore my earlier link to a downloads page, it's not the proper site for Lubuntu. I'll follow up with a proper link.

**Comment by grahamperrin** [comment]
Before a screen recording of Lubuntu, here's an **off-topic** frame from an October screen recording of FreeBSD-CURRENT: 

![image](https://user-images.githubusercontent.com/192271/68360792-4c0dbb80-0119-11ea-86bd-19f6b0133ba0.png)

- home-built Waterfox Classic 56.2.14 (20191007063316). 

I think, it's additionally off-topic because the symptom is **not** consistently reproducible with this operating system. 

Full recording(s) available on request.

**Comment by grahamperrin** [comment]
The long, first recording from around twenty-four hours ago: 

[2019-11-06 05:16 Waterfox Classic, Lubuntu, secure connection failures.zip](https://github.com/MrAlex94/Waterfox/files/3817964/2019-11-06.05.16.Waterfox.Classic.Lubuntu.secure.connection.failures.zip)

- around eleven minutes long. 

I'll outdate this comment and add a shorter recording.

**Comment by grahamperrin** [comment]
> boot the image in live mode and aim to make this issue reproducible in the cleanest possible environment.

Hmm. Lubuntu in live mode (no virtual hard disk), the initial visit failed but stupidly I didn't record the screen. 

Second and subsequent visits succeed: 

[2019-11-07 05:20 Lubuntu live.zip](https://github.com/MrAlex94/Waterfox/files/3817990/2019-11-07.05.20.Lubuntu.live.zip)

Time, now, to tell whether it's still consistently reproducible in the Lubuntu that I normally use for testing …

**Comment by grahamperrin** [comment]
**Highly** (not _consistently)_ reproducible with Lunbutu 19.04 in live mode: 

[2019-11-07 05:43 Waterfox Classic, Lubuntu live.tar.gz](https://github.com/MrAlex94/Waterfox/files/3818080/2019-11-07.05.43.Waterfox.Classic.Lubuntu.live.tar.gz)

This coincides with my discovery that the `lubuntu.net` page was misleading. 19.04 is outdated …

**Comment by grahamperrin** [comment]
Highly reproducible at https://onlinebanking.mtb.com/Enrollment/Enroll with Lubuntu 19.10 https://lubuntu.me/downloads/ in live mode, booted from the `.iso` in a VirtualBox guest. A screen recording: 

- [2019-11-07 06:29 Waterfox Classic, Lubuntu 19.04 live.tar.gz](https://github.com/MrAlex94/Waterfox/files/3818162/2019-11-07.06.29.Waterfox.Classic.Lubuntu.19.04.live.tar.gz)
- https://put.re/player/2bjt1hWn.mp4

First run of Waterfox Classic: 

- 0:43 on the timeline. 

Whilst the issue appears to be _consistently_ reproducible in this particular recording, given previous test results I should cautiously describe it as _highly_ reproducible in this environment. 

```
grahamperrin@momh167-gjp4-8570p:~ % date ; uname -v
Thu  7 Nov 2019 06:23:37 GMT
FreeBSD 13.0-CURRENT #35 r354082: Sat Oct 26 00:25:05 BST 2019     root@momh167-gjp4-8570p:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG 
grahamperrin@momh167-gjp4-8570p:~ % pkg query '%o %v %R' virtualbox-ose virtualbox-ose-kmod
emulators/virtualbox-ose 5.2.34 poudriere
emulators/virtualbox-ose-kmod 5.2.34 poudriere
grahamperrin@momh167-gjp4-8570p:~ % 
```

I'll link to this comment from the Lubuntu subreddit so, for convenience: 

https://www.waterfox.net/releases/ ▶ https://storage-waterfox.netdna-ssl.com/releases/linux64/installer/waterfox-classic-2019.10.en-US.linux-x86_64.tar.bz2 for late October build 20191023220600 of Waterfox Classic 2019.10

– and to put it in context, for a Lubuntu audience, from <https://old.reddit.com/comments/dorwpc/-/f6nkj1j/?context=2>: 

>>> … Possibly. It loads on macOS/Windows so I’ll see if there’s anything Linux specific.

> … the website loads fine on Debian 9 and Debian 10. I'm wondering if there's some incompatibility with your system library version. … 

----

Cross reference https://www.reddit.com/r/Lubuntu/comments/dsu4qo/-/ and **postscript**: 

- for me, the symptoms are now **consistently reproducible** with home-built Waterfox Classic 2019.10 (20191103135811) on **FreeBSD-CURRENT**. 

`waterfox --safe-mode -p test https://onlinebanking.mtb.com/`

Given the 20191103 date of my build, it's _likely_ that I tested M&T Bank pages at least once during the three days prior to today (Wednesday), however this morning is my first observation of the issue being **consistently** reproducible on FreeBSD-CURRENT. In other words, there's still a slight sense of randomness.

**Comment by grahamperrin** [comment]
@hawkeye116477 re: https://www.reddit.com/r/waterfox/comments/dorwpc/-/f5px2gy/ and given the variation across platforms and distributions, I wonder about certificate trust issues etc.. 

----

Probably way off-topic but here goes: recent Mozilla bug [1594257 - CCADB entries generated 2019-11-05T23:30:00Z](https://bugzilla.mozilla.org/show_bug.cgi?id=1594257) includes a list of _Intermediates to be revoked_ – are those intermediate _certificates_? Is it possible that some prior revocation (not necessarily by Mozilla) has affected Waterfox Classic on some platforms/distros more than others? 

Note that I didn't seek bug 1594257 in relation to this issue 1259, I simply stumbled across 1594257 and the security aspect caught my eye. 

----

I'm more inclined to focus on the TLS handshaking observation. Thoughts? 

For as long as I have the symptoms consistently reproducible on FreeBSD-CURRENT, it should be relatively easy for me to get diagnostic information, although I'll need ELI5 guidance – my network/Internet/security fu is very basic. 

TIA

**Comment by MrAlex94** [comment]
Thanks for the links. I’ve been investigating this. Happens when Classic is built with certain toolchains. Have been focusing on the website for now though, will get into it properly when I can.
